src/content/drafts/how-to-create-a-custom-eslint-rule.md

```diff
@@ -0,0 +1,282 @@
+
+For now, let's briefly talk through each of these pieces. I'll start by showing you some code, and then I'll talk through what each line does (you can find the gist for entire file [here](https://gist.github.com/mike-solomon/1f191393b5bef696937409491c8c559b) if that's easier). 
+
+#### Meta Section
+
+For the bacon example, the `meta` section would look like:
+
+```javascript
+meta: {
+  type: 'suggestion',
+
+  docs: {
+    description: "Don't allow bacon in identifiers",
+    recommended: false,
+    url: "https://eslint.org/docs/rules/no-bacon",
+  },
+
+  schema: [], // no options
+
+  messages: {
+    avoidBacon: "The word bacon was found in {% raw %}'{{identifier}}'" {% endraw %}
+  }
+},
+```
+
+Let's discuss some of the keys:
+
+* The `type` key (line 2) is one of `'problem'`, `'suggestion'`, or `'layout'` depending on how you think the rule should be configured. In the bacon example, I set `type` to `suggestion` due to the fact that identifiers with the name of bacon won't actually cause any errors. Rather, I am *suggesting* they make the change for stylistic purposes. 
+* The `recommended` key (line 6) is either `true` or `false` depending on if it should be a default for anyone using the recommended ESLint rules. In the bacon example, I set `recommended` to `false` because this rule should not be a default rule for anyone using ESLint.
+* The `schema` key (line 10) allows you to add options to the rule so that people can configure it in different ways. In the bacon example, I left `schema` as `[]` because there are no options to configure. An example of something that could be added in the future would be flags like `allowBaconInClassNames` or `preventUnderscoredBacon` -- which could then change what the rule does.
+* The `messages` key (line 12) is a way of providing one or more warnings / errors for the user. By defining these here, it makes it so you don't have to copy the same message to many different files or locations. For instance, the tests I wrote above only have to reference the key of `avoidBacon` instead of copying the entire message to that file.
+* In the `avoidBacon` message (line 13), you can see that I have `identifier` surrounded by double brackets. By enclosing a variable name in double brackets like this, I'm telling ESLint that this warning message expects a variable called `identifier` to be passed in. I'll need to add that in the context section below. You can use this to provide better error messages to your users.
+
+If you want to learn more about what flags and settings exist, please read the  [ESLint Rule Basics](https://eslint.org/docs/latest/developer-guide/working-with-rules#rule-basics) guide.
+
+#### Context Object
+
+For the bacon example, the `context` object would look like: 
+
+```javascript
+create(context) {
+  return {
+    Identifier(node) {
```

Comment:
One thing to keep in mind with this approach is that it will also flag something like this:

```js
let a = bacon;
```

In this case, you're not creating an identifier named `bacon`, but rather consuming one that you may not be able to change.

Can you add a note calling this out?

Reply:
Ooh that's a great call. Thanks!